enhancements/console/user-settings.md

```diff
@@ -0,0 +1,134 @@
+---
+title: user-settings
+authors:
+  - "@cvogt"
+reviewers:
+  - TBD
+  - "@bparees"
+  - "@spadgett"
+approvers:
+  - TBD
+  - "@spadgett"
+creation-date: 2020-09-03
+last-updated: 2020-09-03
+status: implementable
+---
+
+# User Settings for OpenShift Console
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The OpenShift Console has many features targeting different personas; from the Administrator to the Developer. There are limited features in Console today that save settings in order to enhance the user experience upon returning to any particular page. These settings are stored in browser local storage and therefore are not accessible in another browser or machine. Many features that could benefit from customization do not provide customization today due to lack of persisted user settings.
+
+By introducing persisted user settings, many new and existing features will be designed with user settings in mind. Providing a richer experience which the user can tailor to their wants.
+
+## Motivation
+
+After talking with a number of customers, they often use OpenShift Console over a number of machines. They spend time setting up the experience in one browser session and computer, to only have to recreate it on the others. This if often the case due to desktop computers vs lab computers vs home computers. This becomes increasingly important as more features tailor the user experience to the individual.
+
+### Goals
+
+- Provide a mechanism for Console features to store and retrieve user settings.
+- Securely store user settings such that access is restricted to the individual user.
+- Save and restore user settings across different machines and browsers.
+
+### Non-Goals
+
+- Determining which features should make use of this functionality over local and session storage.
+
+## Proposal
+
+### Access Control and Storage
+
+A single `ConfigMap` per user will be used to store user settings. The Console will require a new backend service to manage the creation of and access to the `ConfigMap` on behalf of the user.
+
+User settings `ConfigMaps` will be stored in the `openshift-config-managed` namespace.
```

Comment:
@bparees @jwforres Thoughts on this? We've also talked about

1. Putting all preferences in a one config map with sharding if it grows to big.
2. Using a different namespace like openshift-console or openshift-console-settings to prevent polluting openshift-config-managed with a bunch of config maps